oarepo_citations/resources/csl/csl_to_bibtex.py

```
@@ -18,17 +18,29 @@ def map_resource_type_to_bibtex(resource_type):
     return csl_to_bibtex.get(resource_type, "misc")
 
 
+def valid_int(v, lo, hi):
+    try:
+        v = int(v)
+    except (ValueError, TypeError):
+        return False
+    return lo <= v <= hi
+
+
 def create_bibtex_entry(csl_data, id_):
     entry_type = map_resource_type_to_bibtex(csl_data['type'])
     authors = csl_data.get('author', [])
     title = csl_data.get('title')
-    year = None
-    
+    year, month, day = None, None, None
     if 'issued' in csl_data:
-        date_parts = csl_data['issued'].get('date-parts', [])
+        date_parts = csl_data['issued'].get('date-parts')
         if date_parts:
-            year = date_parts[0][0]
-            month = date_parts[0][1]
+            first_part = date_parts[0]
+            if first_part and valid_int(first_part[0], 1, 9999): 
```

confusing variable naming, i'd spread like it was before: year, month, day = (date_parts[0] + (None,))[:3]

---

Then validate each, validation should raise value error => leading to bad request, rather than ignoring the bad part

---

The thing is that we can render bibtex without any date (or invalid date parts) whatsover. That's why we don't need all properties to render BibTeX, and should not raise BadRequest or any exception if any is missing. It is a problem of the dataset/record itself (or our marshmallow/data transformers), which should be fixed.